src/graphql/data/TopTokenQuery.ts

```diff
@@ -0,0 +1,83 @@
+import graphql from 'babel-plugin-relay/macro'
+import { TimePeriod, TokenData } from 'hooks/useExplorePageQuery'
+import { useLazyLoadQuery } from 'react-relay'
+
+import type { TopTokenQuery as TopTokenQueryType } from './__generated__/TopTokenQuery.graphql'
+
+export function useTopTokenQuery(page: number) {
+  const topTokenData = useLazyLoadQuery<TopTokenQueryType>(
+    graphql`
+      query TopTokenQuery($page: Int!) {
+        topTokenProjects(orderBy: MARKET_CAP, pageSize: 50, currency: USD, page: $page) {
+          name
+          tokens {
+            chain
+            address
+            symbol
+          }
+          markets(currencies: [USD]) {
+            price {
+              value
+              currency
+            }
+            marketCap {
+              value
+              currency
+            }
+            fullyDilutedMarketCap {
+              value
+              currency
+            }
+            volume1H: volume(duration: HOUR) {
+              value
+              currency
+            }
+            volume1D: volume(duration: DAY) {
+              value
+              currency
+            }
+            volume1W: volume(duration: WEEK) {
+              value
+              currency
+            }
+            volume1M: volume(duration: MONTH) {
+              value
+              currency
+            }
+            volume1Y: volume(duration: YEAR) {
+              value
+              currency
+            }
+          }
+        }
+      }
+    `,
+    {
+      page,
+    }
+  )
+
+  const topTokens: Record<string, TokenData> =
+    topTokenData.topTokenProjects?.reduce((acc, token) => {
```

Comment:
i don't love changing the shape of the data on the client like this. ideally the API + objects are well designed so we just take them as they are

Reply:
I think that's an unrealistic expectation as this is used across different systems. Converting to an internal data representation is quite common.

Reply:
sometimes, but why do we always index into the first market? should we only fetch one market then `(limit: 1)`? also think this has impact for entity caching as well

Reply:
ah i haven't used graphql much. yes i think we should just align the query properly, agreed.

Reply:
I think we're doing this in all the other queries too (including price charts) so maybe we can update this and the error/loading states in a separate PR?